### PR TITLE
send request by POST with BODY when URI length size is over 8k

### PR DESCRIPTION
**Pull request by htamakos**
This PR is to solve the problem that when the request URL is too long, the endpoint returns 414 and cannot be measured.

If the endpoint is using Nginx with the default configuration, "414 URI Too Long" will be returned if the request URL size exceeds 8k.

It is possible to change the acceptable request URL size by changing a parameter (e.g. http://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers) but it is difficult to determine how much to allow. In some cases, it may not even be possible to change such parameters in the first place.

Therefore, we will address this issue by modifying the ATJ. Specifically, if the request URL size exceeds 8k, the d parameter setting will be excluded from the request URL, and the request will be sent as a POST instead of a GET, with the values set in the request parameters included in the BODY.

## Major Changes
- request method is switched by calculating request URL size
  - Supported by xhr, fetch, and sendBeacon
  - POST requests now include a Blob object with type `application/json` in the Body.

- added `Utils#setSendBeaconStatusForTestUse` method in Utils class
  - This is a method for testing
  - This was inconvenient for testing because sendBeaconStatus is not an instance variable but a variable defined in the module, which cannot be changed externally, and once sendBeaconStatus is false, there is no way to change it. Therefore, we prepared a method to change sendBeaconStatus.
   - The method name "ForTestUse" clearly indicates that it is for testing.

- Added sinon for testing `Utils#transmit` to have a stub for sending requests when testing.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -433,16 +442,21 @@ export default class Utils {
         return r;
     }
 
-    xhr(u, a) {
+    xhr(u, a, m, b) {
```
Comment (komlow) [inline src/utils.js:445]: [note]
mはHTTPメソッド、bはHTTPメソッドをPOSTにした場合のリクエストボディ

**Review by komlow — APPROVED** [review]
LGTM!

**Comment by hjmsano** [comment]
@htamakos ESLint says there are some lack of semi-colons. To fix this, can you merge [this PR](https://github.com/Nikkei/atlas-tracking-js/pull/123) ?

**Comment by hjmsano** [comment]
The minified SDK size will increase +600 bytes

File: src/utils.js [hunk 2]
```diff
@@ -5,6 +5,15 @@ const SDK_VERSION = process.env.npm_package_version;
 const SDK_API_KEY = process.env.SDK_API_KEY || 'test_api_key';
 const DEFAULT_ENDPOINT = process.env.DEFAULT_ENDPOINT || 'atlas.local';
 
+// Nginx URL Length default Limit
+// See: http://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers
+const MAX_REQUEST_URL_LENGTH = 8 * (1 << 10)
```
Comment (hjmsano) [inline src/utils.js:10]: semi-colon required

**Comment by hjmsano** [comment]
LGTM

**Review by hjmsano — APPROVED** [review]

**Pull request merged**